Frontend/src/Components/Graph/CustomGraph.js

```diff
@@ -159,139 +162,117 @@ function getOptions(now, secondsRetained, colorMode) {
   };
 }
 
+let colormap= {};
+function genColor(key) {
+
+  if (key in colormap) {
+    return colormap[key];
+  } else {
+    const color = "#"+Math.floor(Math.random()*16777215).toString(16)
+    colormap[key] = color;
+    console.log(color);
+    return color;
+  }
+}
+
+function getFormattedData(timestamp, dataset) {
+  let res = {labels: timestamp};
+  res['datasets'] = [];
+  for(const key in dataset) {
+    if(key !== 'timestamps')
+      res['datasets'].push({label: key, data: dataset[key], borderColor: genColor(key), backgroundColor: 'rgba(255, 99, 132)'});
+  }
+  return res
+}
+
 /**
- * Creates a customizable graph.
- *
- * @param {any} props the props to pass to this graph; any {@link StackProps} will be passed to the overarching container
- * @param {(name: string, isNew: boolean, data: {datasets: string[], historyLength: number}) => void} props.onSave the callback function for when the user attempts to save this graph
- * @param {string} props.title the title that this graph has
- * @param {string[]} props.initialDatasets a list of the IDs of the initial datasets
- * @param {any} props.packedData the queue of data, packed in Chart.js format, coming from the solar car
- * @param {number} props.secondsRetained the number of seconds to retain a point on this grpah
- * @returns
+ * The graph component, pass in list of data and it will fetch itself
+ * @param props
+ * @constructor
+ */
+function Graph(props) {
```

Comment:
I could be wrong, but I have a sneaking suspicion that this could be defined within the `CustomGraph` component to help reduce the electric boogaloo of things being redefined